build-support/docker/centos7/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+FROM centos:7
+
+# Install a (more) modern gcc, a JDK, and dependencies for installing Python through Pyenv.
+RUN yum -y update
+RUN yum install -y centos-release-scl
+RUN yum install -y \
+        make \
+        devtoolset-7-gcc{,-c++} \
+        git \
+        java-1.8.0-openjdk-devel \
```

Comment:
Probably we should also be standardizing on at least java 11?

Reply:
I don't think we support Java 11! Pants assumes `tools.jar`, which was removed in Java 9 per https://stackoverflow.com/a/53707739.

When running `./pants lint ::` with brew installed Java 12, I get this exception: `Exception message: Failed to locate tools.jar library`.

Reply:
bump on this @benjyw - I imagine you've discovered this recently